lib/t/matches.ex

```diff
@@ -428,19 +428,18 @@ defmodule T.Matches do
 
     broadcast_for_user(mate, {__MODULE__, [:timeslot, :accepted], timeslot})
 
-    case DateTime.compare(slot, reference) do
-      :lt ->
-        now_push =
+    pushes =
+      if DateTime.compare(slot, reference) in [:lt, :eq] do
+        notify_timeslot_started(%Match{id: match_id, user_id_1: picker, user_id_2: mate})
```

Comment:
there is no guarantee `picker < mate`

Reply:
does it matter though?

Reply:
https://github.com/getsince/test3/blob/5e450ee23ca2bc77049d593827087d9b64160f7b/lib/t/matches.ex#L608

Reply:
https://github.com/getsince/test3/blob/5e450ee23ca2bc77049d593827087d9b64160f7b/lib/t/matches.ex#L604:L608

Reply:
yes, maybe a function like `build_match(id, user_ids)` can be added to enforce the rule

it's easier to follow the rule `user_id_1 < user_id_2` everywhere than to fix the consequences of not following it

Reply:
either way, building structs manually like in `%Match{id: match_id, user_id_1: picker, user_id_2: mate}` should be avoided, easy to make mistake

Reply:
in this particular case i'd rewrite add a new clause of `notify_timeslot_started` to also accept match_id and user ids list

Reply:
please add a new clause of notify_timeslot_started to also accept match_id and user ids list and use it from this function